ADD : [네트워크] HTTP & HTTPS - 발표 자료 추가

goretex404 opened the pull request:
네트워크 폴더 생성 및 HTTP & HTTPS 발표자료 추가

goretex404 committed:
ADD : [네트워크] HTTP & HTTPS - 발표 자료 추가

네트워크 폴더 생성 및 HTTP & HTTPS 발표자료 추가

dev-splin commented on Network/HTTP & HTTPS.md:
```diff
@@ -0,0 +1,182 @@
+
+![img](https://mblogthumb-phinf.pstatic.net/MjAxODExMjZfMjQ4/MDAxNTQzMTk5MjkxMjg5.hgyDX0eeJFfTy-LzLODlotK_ebFEMyJlwbxL7c-AkUMg.dVdhZiAGB2DFjXYhH3VNrODn66GgvtZXo_FPSUS0JGsg.PNG.chodahi/image.png?type=w800)
+
+- 암호화 할 때 사용하는 키와 복호화 할 때 사용하는 키가 다른 경우를 말한다. AES
```
AES는 대칭 키 알고리즘으로 알고 있는데, 왜 비대칭 키 설명에서 나온 것일까요??

goretex404 replied:
발표 당시 설명중 RSA와 AES 암호화를 혼동하여 AES라 적어둔 것 같습니다. 해당 문구 삭제토록하겠습니다.

dev-splin commented on Network/HTTP & HTTPS.md:
```diff
@@ -0,0 +1,182 @@
+  타인에게 절대 노출되어서는 안되는 개인키(비밀키)를 토대로 만든 공개키가 쌍을 이룬 형태이다.
+  2개의 키 사용
+   \- 공개키(Public Key) : 사람들에게 공개된 키이며 정보를 암호화 할 수 있다. 
+   \- 개인키(Private Key) : 사용자만 알고 있는 암호를 풀 수 있는 키 
```
공개키와 개인키에 "\\-"로 되어있어서 리스트 생성이 안되는 것 같습니다!

goretex404 replied:
수정 반영토록 하겠습니다 !

dev-splin commented on Network/HTTP & HTTPS.md:
```diff
@@ -0,0 +1,182 @@
+
+- **문제점**
+
+  - 위에서 가정한 조건은 A와 B가 각자의 개인키와 공개키를 가지고 있고
```
A(왼쪽)이 클라이언트고 B(오른쪽)이 서버라면,
A는 B가 제공하는 공개키를 가지고 암호화 하고
B는 개인키(비밀키)로 복호화하는 방식으로 진행되는 것으로 알고 있는데,
그렇다면 A와 B는 각자의 개인키와 공개키를 가지는 것이 아니라 A는 공개키, B는 개인키(비밀키)를 가지는 거 아닌가요??

goretex404 replied:
지적해주신 내용이 맞습니다. 자료 조사시 그림과 내용을 각각 다른 곳에서 가져와 불일치하는 부분이 발생했으며
본문에서는 비대칭키 방식만을 적용하여 쌍방으로 평문을 주고받는 상황을 가정하고 작성 된 내용이라 각자의 개인키와 공개키를 가지고 있다고 설명된것으로 보입니다. 해당 문구를 삭제하는 것이 첨부 그림의 흐름과 더 적합하다고 판단되어 삭제토록하겠습니다.

dev-splin commented on Network/HTTP & HTTPS.md:
```diff
@@ -0,0 +1,182 @@
+   - 인증서를 사용한 웹서버의 URL
+   - 인증서를 발행한 기관 이름
+
+**클라이언트와 서버의 통신 흐름 과정**
```
하이브리드 암호화 방식이라고 명시해주시면 더 좋을 것 같습니다^~^

goretex404 replied:
네 수정토록 하겠습니다!

dev-splin commented on Network/HTTP & HTTPS.md:
```diff
@@ -0,0 +1,182 @@
+
+그렇다면 언제 HTTP를 쓰고, 언제 HTTPS를 쓰는 것이 좋겠는가?
+
+개인 정보와 같은 민감한 데이터를 주고 받아야 한다면 HTTPS를 이용해야 하지만, 단순한 정보 조회 등만을 처리하고 있다면 HTTP를 이용하면 된다.
```
https://tech.ssut.me/https-is-faster-than-http/
발표 때 말씀하신 것 처럼 한재엽님 글에 있는 위 링크와 이에 관한 내용을 조금 첨부해주시면 좋을 것 같습니다!!

goretex404 replied:
좋은 지적 감사드립니다 해당 내용 보충하도록 하겠습니다.

dev-splin reviewed (changes requested):
몇 가지의 궁금증과 글 읽는 사람 입장으로써의 제안을 해봤습니다..!!
리뷰하려고 하이브리드 암호화에 대한 여러 글을 본 결과, 
하이브리드 암호화에 대한 설명 정말 쉽게 잘 해주셨다는 걸 느낄 수 있었습니다! 👍

goretex404 committed:
FIX : [네트워크] HTTP & HTTPS - 발표 자료 일부 수정

이전 PR 코멘트에 수정사항에 대한 내용을 re-comment하였습니다.
1. AES 문구 삭제 ( 설명중 RSA와 AES 암호화를 혼동하여 작성되었음)
2. 공개키 / 개인키 설명 리스트로 생성되지 않는 문제 수정
3. 비대칭키 문제점 부분에서 그림과 매칭되지 않는 설명문구 삭제
4. SSL의 암호화 알고리즘 설명 문구 추가 및  SSL 통신 흐름 과정에 하이브리드 암호화라 명시문구 추가
5. HTTP vs HTTPS 부분 설명 보충

goretex404 commented on Network/HTTP & HTTPS.md:
```diff
@@ -87,16 +87,14 @@ HyperText Transfer Protocol over Secure Socket Layer, HTTP over TLS, HTTP over S
 ![img](https://mblogthumb-phinf.pstatic.net/MjAxODExMjZfMjQ4/MDAxNTQzMTk5MjkxMjg5.hgyDX0eeJFfTy-LzLODlotK_ebFEMyJlwbxL7c-AkUMg.dVdhZiAGB2DFjXYhH3VNrODn66GgvtZXo_FPSUS0JGsg.PNG.chodahi/image.png?type=w800)
 
-- 암호화 할 때 사용하는 키와 복호화 할 때 사용하는 키가 다른 경우를 말한다. AES
+- 암호화 할 때 사용하는 키와 복호화 할 때 사용하는 키가 다른 경우를 말한다.
```
1. AES 문구 삭제

goretex404 replied:
1. AES 문구 삭제 ( 설명중 RSA와 AES 암호화를 혼동하여 작성되었음)

goretex404 commented on Network/HTTP & HTTPS.md:
```diff
@@ -87,16 +87,14 @@ HyperText Transfer Protocol over Secure Socket Layer, HTTP over TLS, HTTP over S
-   \- 공개키(Public Key) : 사람들에게 공개된 키이며 정보를 암호화 할 수 있다. 
-   \- 개인키(Private Key) : 사용자만 알고 있는 암호를 풀 수 있는 키 
+   - 공개키(Public Key) : 사람들에게 공개된 키이며 정보를 암호화 할 수 있다. 
+   - 개인키(Private Key) : 사용자만 알고 있는 암호를 풀 수 있는 키 
```
2. 공개키 / 개인키 설명 리스트로 생성되지 않는 문제 수정

goretex404 commented on Network/HTTP & HTTPS.md:
```diff
@@ -87,16 +87,14 @@ HyperText Transfer Protocol over Secure Socket Layer, HTTP over TLS, HTTP over S
 - **문제점**
 
-  - 위에서 가정한 조건은 A와 B가 각자의 개인키와 공개키를 가지고 있고
-
```
3. 비대칭키 문제점 부분에서 그림과 매칭되지 않는 설명문구 삭제

goretex404 commented on Network/HTTP & HTTPS.md:
```diff
@@ -125,7 +125,7 @@ SSL 프로토콜은 Netscape 사에서 웹 서버와 브라우저 사이의 보
    - 인증서를 발행한 기관 이름
 
-**클라이언트와 서버의 통신 흐름 과정**
+**클라이언트와 서버의 통신 흐름 과정 (SSL - 하이브리드 암호화 방식)**
```
4. SSL의 암호화 알고리즘 설명 문구 추가 및  SSL 통신 흐름 과정에 하이브리드 암호화라 명시문구 추가
